lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -141,7 +146,23 @@ class Output
         }                                                                     \
     } while (false)
 
-#define UNWRAP_RESOURCE(resource, innerType) reinterpret_cast<PlatformAgnosticIntlObject<innerType> *>(resource)->GetInstance();
+// Some ICU functions don't like being given null/0-length buffers.
+// We can work around this by making a local allocation of size 1 that will still
+// trigger a U_BUFFER_OVERFLOW_ERROR but will allow the function to return the required length.
+#define ICU_FIXBUF(type, bufArg, bufArgLen)                                   \
+    type __temp; \
+    const int __tempLen = 1; \
+    bool isTemporaryBuffer = false; \
```

Comment:
I'd feel better about these hidden locals if we had a pair of macros that put all of this inside a scope (as we do elsewhere in the runtime)

```
#define ICU_FIXBUF_BEGIN(...) \
{ /* open brace to start a scope */ \
/*  ... [ this implementation code ] ... */

#define ICU_FIXBUF_END() } /* close brace to semantically close the scope */
```

Reply:
Or alternatively, if isTemporaryBuffer is the only one that should be visible outside this macro, declare it outside the macro, and let the macro-internal declarations be inside a scope inside this macro definition so their scope doesn't leak.

---
In reply to: [155905911](https://github.com/Microsoft/ChakraCore/pull/4390#discussion_r155905911) [](ancestors = 155905911)

Reply:
I feel like that is somewhat overkill for now, since all of these functions are so short. We also need their lifetime to be until the call to the ICU function returns, so it would be wrapping most of the function in them